### ISSUE TITLE
RigidBody3D: Applying a force at a position in _physics_process results in incorrect torque

### ISSUE THREAD

**Issue by Rylius** [godotengine/godot#85865]
### Tested versions

- Reproducible in v4.2.stable.mono.official [46dc27791]
- Reproducible in v4.1.3.stable.mono.official [f06b6836a]
- Reproducible in v4.0.3.stable.mono.official [5222a99f5]

Also reproducible with godot-jolt, if that helps. (The MRP uses Godot physics.)

### System information

Godot v4.2.stable.mono - Windows 10.0.19045 - GLES3 (Compatibility) - NVIDIA GeForce RTX 3070 (NVIDIA; 31.0.15.3623) - AMD Ryzen 7 2700X Eight-Core Processor (16 Threads)

### Issue description

I'm using `apply_force(Vector3, Vector3)` (where the second parameter is *not* zero) in `_physics_process` to lift a rigid body (counter-acting gravity and adding a slight forward motion). The angular velocity rapidly increases and makes the body spin out of control.

See green (first) and blue (third) cubes here:

https://github.com/godotengine/godot/assets/1206571/0bf2391f-240e-4a12-b1ad-88ec38db87ac

- 1/green: GDScript + `apply_force`
- 2/red: GDScript + `apply_central_force` + `apply_torque`
- 3/blue: C# + `ApplyForce`
- 4/yellow: C# + `ApplyCentralForce` + `ApplyTorque`

### Steps to reproduce

```gdscript
extends RigidBody3D


func _physics_process(_delta: float) -> void:
	var force := Vector3.UP.rotated(Vector3.FORWARD, deg_to_rad(5.0)) * 110.0
	var target_position := global_transform * (Vector3.UP * 0.5)

	apply_force(force, target_position)
```

My grasp on physics is limited but I'm pretty sure this should result in the bodies *slightly* swinging around the point where the force is applied (as is the case with the red/second and yellow/fourth cubes in the video).

I took a quick look in the engine source and found the `apply_force` function: https://github.com/godotengine/godot/blob/2f73a059cefadcd944b6874f2557ec82e46a562d/servers/physics_3d/godot_body_3d.h#L255-L258

This seems to be a nice shortcut for `apply_central_force` and `apply_torque`, so I figured I'd try those manually:

```gdscript
extends RigidBody3D


func _physics_process(_delta: float) -> void:
	var force := Vector3.UP.rotated(Vector3.FORWARD, deg_to_rad(5.0)) * 110.0
	var target_position := global_transform * (Vector3.UP * 0.5)

	apply_central_force(force)
	apply_torque((target_position - (global_transform * center_of_mass)).cross(force))
```

And that works as expected!

Please let me know if I'm misunderstanding something or doing incorrect calculations here (not ruling it out).

--

A few notes about the MRP:
- Press R to reset the bodies.
- Inertia is set to 100, 100, 100. This slows down the spinning and makes the bodies a bit more stable but does not affect the issue.
- The Linear Z and Angular X/Y axes on the rigid bodies are locked. This does not affect the issue.
- Tried it with Godot Physics and Jolt. No difference.

--

For some extra fun: C# also exhibits this behavior, but it somehow spins in the other direction and is faster.

### Minimal reproduction project (MRP)

[force-torque-mrp.zip](https://github.com/godotengine/godot/files/13591075/force-torque-mrp.zip)
[force-torque-mrp-csharp.zip](https://github.com/godotengine/godot/files/13591096/force-torque-mrp-csharp.zip)

**Comment by rohanrichards** [comment]
I don't think you got around to updating the docs, that or they were reverted, I just lost a day to this one 😂 @Rylius

**Comment by Rylius** [comment]
Never got around to it, sorry.

Current:
```
				Applies a positioned force to the body. A force is time dependent and meant to be applied every physics update.
				[param position] is the offset from the body origin in global coordinates.
```

Proposed:
```
				Applies a positioned force to the body. A force is time dependent and meant to be applied every physics update.
				[param position] is the relative offset from the body origin in global coordinates.
```

Don't feel like that quite cuts it, but also not quite sure what to change. Any ideas?
(I don't mind preparing a PR, just need a bit of input first)

cc @rohanrichards

**Comment by rohanrichards** [comment]
I think I would have still gotten tripped up on the "global coordinates." Without an understanding that this is referring to a coordinate system, not global space location, I think it's still hard to understand. Without a deeper grasp on it, I have to wonder how important it is to even specify global coordinates. Maybe just expressing "Local space" explicitly might be enough:

[param position] is the local offset from the body origin.

**Comment by mithost** [comment]
> After a good night's sleep I figured out what's going and it's entirely my fault. `apply_force` expects the position in local space, not global:
> 
> 	var target_position := transform.basis * (Vector3.UP * 0.5)
> 
> 	apply_force(force, target_position)
> 
> (It's important to use `transform.basis`, the transform as a whole also adds translation which we don't want here.)
> 
> I'll see if I can get the documentation updated; I'm probably not the only one confused by "position is the offset from the body origin in global coordinates".

I feel like the documentation for `apply_force()` should include the codeblock provided in this comment as an example, as I too have lost a day to this and this github page only appeared a few pages deep into web search results.
